calendar_reboot.js

```diff
@@ -66,44 +65,54 @@ $(document).ready(function() {
         //with the class form-group, also adds the hidden class on the span
         //with class sr-only (for screen readers)
         
-        id.removeClass('has-error has-feedback');
-        srId.removeClass('hidden');
+        $(id).removeClass('has-error has-feedback');
+        $(srId).removeClass('hidden');
     };
     
     var addGlyphicon = function(id) {
         //removes the hidden class to the glyphicon tag with id = id
-        id.removeClass('hidden');
+        $(id).removeClass('hidden');
     };
     
     var removeGlyphicon = function(id) {
         //adds the hidden class to the glyphicon tag with id=id
-        id.addClass('hidden');
+        $(id).addClass('hidden');
     };
     
-    validateForm = function(startDate, endDate) {
-        //check that endDate comes after startDate
-        //checks 
-          //if there is no valid startDate, prompt user for a valid startDate
+    var removeFormErrors = function() {
+        //remove the error classes and glyphicons from the form inputs
+        removeFormError('#dateFormGroup', '#inputError-dateFormGroup');
+        removeFormError('#dateFormGroup2', '#inputError-dateFormGroup2');
+        removeFormError('#titleFormGroup', '#inputError-titleFormGroup');
+        removeGlyphicon('#span-titleFormGroup');
+    };
+    
+    var validateDates = function(startDate, endDate) {
+        console.log("validating the dates");    
         
-        var startDateMoment = moment(startDate, "YYYYMMDD");
-        var endDateMoment = moment(endDate, "YYYYMMDD");
+        var startDateMoment = moment(startDate, "YYYY-MM-DD");
+        var endDateMoment = moment(endDate, "YYYY-MM-DD");
         
         if (startDateMoment.isBefore(endDateMoment)) {
+            removeFormError('#dateFormGroup2', '#inputError-dateFormGroup2');
             return true;
         }
         else {
-            console.log("endDate is before startDate");
+            alert("The end date needs to be after the start date.");
```

Comment:
Got rid of the alert. Added help text below the end date input field that says "The end date must be after the start date." The whole input box turns red if the user provides an invalid date (a date before the start date or no input at all or null).